functions/messages.js

```diff
@@ -0,0 +1,147 @@
+/* eslint-disable @typescript-eslint/no-var-requires */
+const fetch = require('node-fetch');
+const {
+  CreateContactError,
+  CreateConverstaionError,
+  CreateLabelError,
+  CreateMessageError,
+  RecaptchaError
+} = require('./errors');
+const ChatwootMessenger = require('./messenger');
+const { reportError } = require('./error-logger');
+
+const RECAPTCHA_SECRET = process.env.GATSBY_RECAPTCHA_V3_SECRET_KEY;
+const RECAPTCHA_MINIMUM_SCORE = 0.5;
+
+const verifyRecaptcha = async (token) => {
+  try {
+    const response = await fetch(
+      `https://www.google.com/recaptcha/api/siteverify?secret=${encodeURIComponent(
+        RECAPTCHA_SECRET
+      )}&response=${encodeURIComponent(token)}`,
+      {
+        method: 'POST'
+      }
+    );
+    const data = await response.json();
+
+    return data.success && Number(data.score) >= RECAPTCHA_MINIMUM_SCORE;
+  } catch (error) {
+    throw new RecaptchaError();
+  }
+};
+
+exports.handler = async (event) => {
+  if (event.httpMethod !== 'POST') {
+    return {
+      statusCode: 405,
+      body: JSON.stringify({
+        message: 'Method Not Allowed',
+        status: 405
+      })
+    };
+  }
+
+  if (!event.body) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({
+        type: 'Missing parameters',
+        message: 'Something went wrong, please try again',
+        status: 400
+      })
+    };
+  }
+
+  const {
+    email,
+    name,
+    message,
+    subject,
+    'g-recaptcha-response': recaptchaToken
+  } = JSON.parse(event.body || {});
+  const messenger = new ChatwootMessenger(email, name, message, subject);
+
+  let statusCode = 500;
+  let body = JSON.stringify({
+    message: 'Internal Error'
+  });
+
+  if (!email || !name || !message || !subject) {
+    return {
+      statusCode: 400,
+      body: JSON.stringify({
+        type: 'Missing parameters',
+        message: 'Something went wrong, please try again',
+        status: 400
+      })
+    };
+  }
+
+  try {
+    const isValidRecaptcha = await verifyRecaptcha(recaptchaToken);
+
+    if (!isValidRecaptcha) throw new RecaptchaError();
+
+    await messenger.createContact();
+    await messenger.createConverstaion();
+    await messenger.createLabel();
+    const response = await messenger.createMessage();
+
+    statusCode = 200;
+    body = JSON.stringify({
+      content: message,
+      id: response.id,
+      message: 'Message created successfully',
+      status: 200
+    });
+  } catch (error) {
+    await reportError(error);
+
+    if (error instanceof RecaptchaError) {
```

Comment:
BTW, have you considered enriching your custom errors instead of making this chain of "IF" statements? Perhaps, you could do something like:

```js
class CreateMessageError extends Error {
    constructor(message) {
        super(message);
        this.name = "CreateMessageError";
    }
}
```

and then

```js
try {
    /* ... */
} catch (e) {
    return {
        body: JSON.stringify({
            type: e.name,
            message: e.message,
            statusCode: 409    
        })
        statusCode: 409
    }
}
```

It would be easier to maintain and while you are editing the custom Error class you still are using the Error expected structure. The only thing I see is that IDK why we are returning the status code twice. Maybe a small mistake?

Reply:
@duranmla well, being honest with you, I don't believe catching errors with an `if clause` will make this feature harder to maintain, nevertheless, I do believe, we need to improve the messages

Reply:
2.- We are not returning the status code twice. The status code is stored into a variable called `statusCode` and its default value is `500`. If the Recaptcha fails, I'm re-assigning the variable to be 409